projects/FinalProject-Bugs/dominion/unittest7.c

```diff
@@ -0,0 +1,244 @@
+#include "dominion.h"
+#include "dominion_helpers.h"
+#include <stdio.h>
+#include "rngs.h"
+#include <stdlib.h>
+#include <string.h>
+
+int main() {
+
+   // set players and seed
+   int numOfPlayers = 2;
+   int seed = 232;
+
+   // declare game state
+   struct gameState G, testGame;
+
+   // set card array
+   int k[10] = {adventurer, baron, gardens, embargo, village, minion, mine, cutpurse, sea_hag, tribute};
+
+   // initialize game
+   initializeGame(numOfPlayers, k, seed, &G);
+
+   int currPlayer = 0;
+   int nextPlayer = 1;
+
+   // begin testing bug 7 - tribute
+   // reveal two different treasure cards
+   printf("--------------------------------------------------------------------------\n");
+   printf("BEGIN TEST #1 FOR BUG 7\n\n");
+
+   G.hand[currPlayer][0] = baron;
```

Comment:
Could assign the hand like this:
```
int playerHand[5] = { baron, duchy, mine, minion, ambassador } ;
memcpy(G.hand[currPlayer], playerHand, sizeof(int) * 5);
```
I think it's generally a good idea to avoid manual assignment of arrays by index, especially if it's many of them, because it could be error-prone.